### ISSUE TITLE
encrypts class method may conflict with Rails-native encryption

### ISSUE THREAD

**Issue by benjaminoakes** [ankane/lockbox#107]
https://github.com/rails/rails/pull/41659 adds encryption natively in Rails.  Here is an example:

```ruby
class Person < ApplicationRecord
  encrypts :name
end
```

This naming would seem to conflict with Lockbox's `encrypts` method.  Are there plans in motion to handle this?

**Comment by ankane** [comment]
Hey @benjaminoakes, currently thinking of changing it to `encrypt`.

**Comment by benjaminoakes** [comment]
Thank you for your response.

That name would certainly not conflict, but it may also be unclear.  A name that includes `lock` or `box` (or both) would help clarify, especially if anyone chooses to transition which library they're using and they have both `encrypts` and `encrypt` in a single repository.

One alternative that seems natural:

```ruby
lockbox_encrypts :name
```

**Comment by masterkain** [comment]
attr_encrypted, attr_locked, attr_lockboxed? unsure 😊

**Comment by ankane** [comment]
Added a `lockbox_encrypts` method that users on Rails edge can use, but the name may change before Rails 7 is officially supported.

**Comment by benjaminoakes** [comment]
[The `lockbox_encrypts` method](https://github.com/ankane/lockbox/blob/75dae4f797136237a76a8bdf3333fbe3e80e4cd5/lib/lockbox/model.rb#L3) has been released [in v0.6.4](https://github.com/ankane/lockbox/commit/75dae4f797136237a76a8bdf3333fbe3e80e4cd5).

**Comment by benjaminoakes** [comment]
Relevant [changelog](https://github.com/ankane/lockbox/blob/master/CHANGELOG.md) entry for v1.0.0:

> - Deprecated `encrypts` in favor of `has_encrypted` to avoid conflicting with Active Record encryption
> - Deprecated `lockbox_encrypts` in favor of `has_encrypted`
